reporter/otlp_reporter.go

```diff
@@ -226,20 +240,25 @@ func (r *OTLPReporter) ReportTraceEvent(trace *libpf.Trace, meta *TraceEventMeta
 		containerID:    containerID,
 	}
 
-	if events, exists := (*traceEventsMap)[key]; exists {
+	traceEventsMap := r.traceEvents.WLock()
```

Comment:
No need to hold the lock while looking up the Cgroupv2 ID and creating the key. So I moved fetching the lock to this later point.